### ISSUE TITLE
Group Bug (BB #823)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#823]
This issue was migrated from bitbucket.
**Original Reporter:** smoldar
**Original Date:** 02.03.2015 13:05:22 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/823

<hr>

1.BUG
When Player A   invites  player B and player C at same time seccond player not getting Accept / Decline invitation, but gets invitation
This i dont know how to fix.
# 

2.BUG
Seccond bug of Groups is in Battlegrounds, you can invite players into BG group...
FIX: GroupHandler.cpp

```
#!c++

void WorldSession::HandleGroupInviteOpcode(WorldPacket & recv_data)
{
    std::string membername;
    recv_data >> membername;
+
+    if (_player->InBattleGround())
+   {
+      SendPartyResult(PARTY_OP_INVITE, membername, PARTY_RESULT_INVITE_RESTRICTED);
+        return;
+    }

    // attempt add selected player
```

And this i dont know 100% that this is right :)

```
#!c++

    Player* leader = objmgr.GetPlayer(group->GetLeaderGUID());
+
+    if(leader && leader->InBattleGround())
+    {
+        SendPartyResult(PARTY_OP_INVITE, "", PARTY_RESULT_INVITE_RESTRICTED);
+        return;
+    }
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 02.03.2015 13:32:50 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/823/#comment-16030149

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 02.03.2015 13:33:14 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/823/#comment-16030155

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 04.03.2015 15:56:28 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/823/#comment-16088475

<hr>

About the first one if A invites B and C...

C gets invitation - he can accept/decline
B gets just invited?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 04.03.2015 15:58:53 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/823/#comment-16088519

<hr>
1. A invites B and after C
2. B gets normally invite, but he must wait and not click on Accept/Decline while A sending next invite
3. C gets invite only for core(cannot by invited by other players) but he dont see Invitation accept/decline

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 05.04.2015 04:30:46 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/823/#comment-17166367

<hr>

Confirmed

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 08.04.2015 19:46:36 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/823/#comment-17252936

<hr>

Fix #823

→ <<cset e48c77000da7>>
